src/Core/CommandBus/ExecutedCommandRegistry.php

```diff
@@ -84,16 +90,42 @@ public function register($command, $handler)
     /**
      * @return array
      */
-    public function getExecutedCommands()
+    public function getExecutedCommands(): array
     {
         return $this->registry['commands'];
     }
 
     /**
      * @return array
      */
-    public function getExecutedQueries()
+    public function getExecutedQueries(): array
     {
         return $this->registry['queries'];
     }
+
+    /**
+     * Returns the file and line that invoked the handle method
+     *
+     * @return array
+     */
+    private function getTrace(): array
+    {
+        $trace = debug_backtrace(DEBUG_BACKTRACE_IGNORE_ARGS, 10);
+
+        foreach ($trace as $step) {
+            if ($step['class'] === TacticianCommandBusAdapter::class
```

Comment:
Did you test for commands calling other commands?

Reply:
No but theoretically it should work because `class` contains the callee, not the caller.

Reply:
I trust you!